Lib/test/test_timeit.py

```diff
@@ -359,11 +359,27 @@ def autorange(self, seconds_per_increment=1/1024, callback=None):
         t = timeit.Timer(stmt=self.fake_stmt, setup=self.fake_setup, timer=timer)
         return t.autorange(callback)
 
+    def autorange_with_argument(self, total_time, seconds_per_increment=1/1024,
+            callback=None):
```

Comment:
Instead of adding a new method, I would simply add the *total_time* to the `autorange` method defined above.